AGENTS.md

```
@@ -355,6 +363,25 @@ client := &http.Client{Timeout: defaults.HTTPClientTimeout}
 resp, err := client.Do(req)
 ```
 
+**Bound response bodies before `io.ReadAll`.** Outbound `io.ReadAll(resp.Body)` is unbounded by default; a hostile or buggy server can exhaust memory. Wrap with `io.LimitReader` against a `pkg/defaults` cap and reject anything that exceeds it:
+```go
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)